operators/ip_match_from_dataset.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2022 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package operators
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/corazawaf/coraza/v3"
+	engine "github.com/corazawaf/coraza/v3"
+)
+
+type ipMatchFromDataset struct {
+	ip *ipMatch
+}
+
+func (o *ipMatchFromDataset) Init(options coraza.RuleOperatorOptions) error {
+	data := options.Arguments
+	dataset, ok := options.Datasets[data]
+	if !ok || len(dataset) == 0 {
```

Comment:
I added the len(dataset) == 0 check and the `TestIpMatchFromEmptyDataset` test to be sure that an existing but empty dataset is not used.
I tested the following test locally, and it is triggered. So a config like ``"SecDataset test `\n`\n"`` leads to an existing, but empty dataset.
What is the expected behaviour? Do we want to trigger an error or allow to load rules without entries?

```
func TestSecDatasetEmpty(t *testing.T) {
  waf := coraza.NewWaf()
  p, _ := NewParser(waf)
  if err := p.FromString("" +
    "SecDataset test `\n`\n"); err != nil {
    t.Error(err)
  }
  if _, ok := p.options.Datasets["test"]; ok {
    t.Error("Added empty dataset")
  }
}
```

Reply:
Good question, I would just leave like that and create an issue to maintain the discussion